lib/Basic/Platform.cpp

```diff
@@ -313,3 +314,37 @@ llvm::Triple swift::getTargetSpecificModuleTriple(const llvm::Triple &triple) {
   return triple;
 }
 
+Optional<llvm::VersionTuple>
+swift::getSwiftRuntimeCompatibilityVersionForTarget(const llvm::Triple &Triple){
+  unsigned Major, Minor, Micro;
+  
+  if (Triple.isMacOSX()) {
+    Triple.getMacOSXVersion(Major, Minor, Micro);
+    if (Major == 10) {
```

Comment:
I think that you can write this more concisely as:

```
  if (Major == 10 && Minor <= 14)
    return llvm::VersionTuple(5, 0);
  return None;
```

Reply:
I'm writing this with an eye toward it eventually growing more arms. There might be more macOS versions with Swift in them in the near future, but who knows